packages/spacecat-shared-data-access/test/unit/models/api-key.test.js

```diff
@@ -149,5 +149,42 @@ describe('ApiKey Model tests', () => {
     it('fails to update deletedAt with invalid date', () => {
       expect(() => apiKey.updateDeletedAt('invalid-date')).to.throw('Invalid deletedAt during update: invalid-date. Must be a valid ISO 8601 date string');
     });
+
+    it('fails to update expiresAt with invalid date', () => {
+      expect(() => apiKey.updateExpiresAt('invalid-date')).to.throw('Invalid expiresAt during update: invalid-date. Must be a valid ISO 8601 date string');
+    });
+
+    it('fails to update revokedAt with invalid date', () => {
+      expect(() => apiKey.updateRevokedAt('invalid-date')).to.throw('Invalid revokedAt during update: invalid-date. Must be a valid ISO 8601 date string');
+    });
+
+    it('returns true if deletedAt, revokedAt, and expiresAt are not set', () => {
+      apiKey.state.deletedAt = null;
+      apiKey.state.revokedAt = null;
+      apiKey.state.expiresAt = null;
+      expect(apiKey.isValid()).to.be.true;
+    });
+
+    it('returns false if deletedAt is after the current time', () => {
+      apiKey.updateDeletedAt(new Date(Date.now() + 10000).toISOString());
+      expect(apiKey.isValid()).to.be.false;
+    });
+
+    it('returns false if revokedAt is after the current time', () => {
+      apiKey.updateRevokedAt(new Date(Date.now() + 10000).toISOString());
+      expect(apiKey.isValid()).to.be.false;
+    });
+
+    it('returns false if expiresAt is after the current time', () => {
+      apiKey.updateExpiresAt(new Date(Date.now() + 10000).toISOString());
+      expect(apiKey.isValid()).to.be.false;
+    });
+
+    it('returns true if deletedAt, revokedAt, and expiresAt are before the current time', () => {
```

Comment:
This is the core of the issue, I think. 

- if expiresAt is in the future: still valid
- if deletedAt or revokedAt are in the past: no longer valid

Reply:
In other words: expiresAt is the only property we should _expect_ to be in the future. If ANY of these properties are set to a date before `now`, the key should not be considered valid.